### PR TITLE
Handle the response shown to the user with correct message

### LINKED ISSUE #5457: [Bug]: Attempt to read from null error in Explore -> Map fragment
### Summary

This issue is caused when there are no elements in `mediaList` Maybe the issue caused when there are no media (images) available at specific coordinate.

### Steps to reproduce

1. Open Explore -> Map Fragment in App
2. Go to Bihta, Patna, India or Coordinates = 25.5303740, 84.8597150 in map 

### Expected behaviour

Place's images should appear on the map.

### Actual behaviour

Toast message `Error fetching nearby places. Attempt to read from null... `

### Device name

Samsung S21

### Android version

Android 14

### Commons app version

4.2.1-debug-HEAD~9be8df3

### Device logs

`java.lang.NullPointerException: Attempt to read from null array
                                                                                                    	at fr.free.nrw.commons.explore.map.ExploreMapPresenter.getLatLngBounds(ExploreMapPresenter.java:223)
                                                                                                    	at fr.free.nrw.commons.explore.map.ExploreMapPresenter.updateMapMarkers(ExploreMapPresenter.java:193)
                                                                                                    	at fr.free.nrw.commons.explore.map.ExploreMapFragment.updateMapMarkers(ExploreMapFragment.java:433)
                                                                                                    	at fr.free.nrw.commons.explore.map.ExploreMapFragment.lambda$populatePlaces$5$fr-free-nrw-commons-explore-map-ExploreMapFragment(ExploreMapFragment.java:413)`

### Screen-shots

https://github.com/commons-app/apps-android-commons/assets/114223204/3babe197-7160-4de2-8760-7035c0676d56

### Would you like to work on the issue?

Yes

### PR DESCRIPTION
**Pull request by rohit9625**
**Description (required)**

Fixes #5457

What changes did you make and why?
I added a new method showLongSnackbar in the ViewUtil.java file to show the user a Snackbar.
Also, updated the showErrorMessage() method in ExploreMapFragment() to use Snackbar instead of a Toast.

**Tests performed (required)**

Tested prodDebug on Samsung A14 with API level 34.

**Screenshots (for UI changes only)**
Screenshot that shows the snackbar:
300532644-5a500990-a801-4ecc-8750-07d666f8cbcf

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java [hunk 1]
```diff
@@ -32,6 +38,36 @@ public static void showShortSnackbar(View view, int messageResourceId) {
             }
         });
     }
+    public static void showLongSnackbar(View view, String text) {
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java:41]: Is this function used in this PR?

Comment (rohit9625) [reply app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java:41]: Oh!! my bad,
I added the implementation in another branch before. 
Just committed the changes to this one :)

File: app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java [hunk 2]
```diff
@@ -447,13 +447,15 @@ public void populatePlaces(LatLng curLatLng) {
...
                     mediaList = explorePlacesInfo.mediaList;
+                    if(mediaList == null) {
+                        showErrorMessage(getString(R.string.no_pictures_in_this_area));
+                    }
                     updateMapMarkers(explorePlacesInfo);
                     lastMapFocus = new GeoPoint(curLatLng.getLatitude(), curLatLng.getLongitude());
                 },
                 throwable -> {
                     Timber.d(throwable);
-                    showErrorMessage(getString(R.string.error_fetching_nearby_places)
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java:455]: I think this one should probably stay as the error for fetching nearby places - or we should understand what these errors are better, given I think this occurs e.g. when it tries to fetch places and the server is down, or there is no connection.

Comment (rohit9625) [reply app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java:455]: Got it 👍,
I was thinking that too. Here we should display the actual message to the user like it was doing before, right?

Comment (domdomegg) [reply app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java:455]: Yes I think so

Comment (rohit9625) [reply app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java:455]: Okay :)
I am making the changes.

**Review by domdomegg — APPROVED** [review]

**Pull request merged**